python/delta_sharing/rest_client.py

```diff
@@ -294,6 +311,30 @@ def query_table_metadata(self, table: Table) -> QueryTableMetadataResponse:
                 metadata=Metadata.from_json(metadata_json["metaData"]),
             )
 
+    @retry_with_exponential_backoff
+    def autoresolve_query_format(self, table: Table):
+        self.set_delta_format_header()
+
+        with self._get_internal(
+            f"/shares/{table.share}/schemas/{table.schema}/tables/{table.name}/metadata",
+            return_headers=True
+        ) as values:
+            headers = values[0]
+            print(headers)
+            # it's a bug in the server if it doesn't return delta-table-version in the header
+            if "delta-table-version" not in headers:
+                raise LookupError("Missing delta-table-version header")
+            # it's a bug in the server if it doesn't return responseformat in the header
```

Comment:
This does not seem right, since it is optional.
See: https://github.com/delta-io/delta-sharing/blob/main/PROTOCOL.md#query-table-metadata

Reply:
yup "delta-table-version" is guaranteed but not "client-reader-features", changed it to autoresolve to parquet if this is missing